src/sumo_backend/mysql.clj

```diff
@@ -0,0 +1,205 @@
+(ns sumo-backend.mysql)
+(require '[clojure.java.jdbc :as jdbc])
+(require '[honeysql.core :as sql]
+         '[honeysql.helpers :refer :all :as helpers])
+(require '[cheshire.core :refer :all]) ; parses json
+(require '[jdbc.pool.c3p0 :as pool])
+
+;; Namespace that connects to MySql
+
+;; reading in keys from an env variable file for now
+;; Or cached:
+;; (def mysql-db
+;;   (memoize
+;;     (fn [] ;; To support multiple DBs, pass in DB name as an arg
+;;       (json/parse
+;;         (slurp ...)
+;;         true)))))
+(def mysql-db 
+  (:local 
+    #_:clj-kondo/ignore
+    (parse-string
+      (slurp "./keys/mysql.json")
+      true)))
+
+;; TODO manage a pool of connections to the DB
+;; Rather than opening a new connection for each query, it's more efficient
+;; and easier on the DB to maintan one or more connections, scaling
+;; them as needed. Depending on how much time you want to spend on this
+;; part, you could either import a library or build it yourself.
+;; There are libraries like c3p0 that do this, which are nice until they
+;; fail, then can be a pain to debug.
+
+;; You could also make an atom or ref that holds active DB connections
+;; and threads running clojure.core.async/go-loops using those connections
+;; to execute SQL queries they read from a clojure.core.async/chan, with
+;; some other thread periodically checking the status of the query chan
+;; and scaling connections/threads accordingly.
+
+;;;;;;;;;;;;;;;;;;;;;;;
+;;; Helper functions
+;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn in?
+  "true if collection contains elm"
+  [coll elm]
+  (some #(= elm %) coll))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Unpaginated Queries
+;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn get-all-ranks-in-tournament
+  "returns all ranks competing in a tournament"
+  [{:keys [month year]}] ; hash set of every rank
+  (into #{} ; how can this be done in one sql call?
```

Comment:
```
(set
  (mapcat
    #(vals (select-keys % [:east_rank :west_rank]))
    (jdbc/query
      mysql-db
      (sql/format
        (sql/build
          :select [:east_rank :west_rank]
          :from :bout
          :where
          [:and
            [:= :year year]
            [:= :month month]])))))
```

Reply:
very nice @bslawski! I also didn't need that `distinct` in those sql selects with the set. and `(into #{}` is just a more verbose way to say `(set` ?